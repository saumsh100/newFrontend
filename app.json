  {
  "name": "carecru",
  "scripts": {
  },
  "env": {
    "NODE_ENV": {
      "required": true
    },
    "RETHINKDB_AUTHKEY": {
      "required": true
    },
    "RETHINKDB_DB": {
      "required": true
    },
    "RETHINKDB_HOST": {
      "required": true
    },
    "RETHINKDB_PORT": {
      "required": true
    },
    "VENDASTA_API_KEY": {
      "required": true
    },
    "VENDASTA_API_USER": {
      "required": true
    },
    "TWILIO_ACCOUNT_SID": {
      "required": true
    },
    "TWILIO_AUTH_TOKEN": {
      "required": true
    },
    "TWILIO_NUMBER": {
      "required": true
    }
  },
  "formation": {
    "web": {
      "quantity": 1
    }
  },
  "addons": [

  ],
  "buildpacks": [
    {
      "url": "heroku/nodejs"
    }
  ]
}
