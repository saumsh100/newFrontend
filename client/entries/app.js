
import '../apps/default';
import '../apps/Dashboard';
