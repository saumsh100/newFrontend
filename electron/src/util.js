
module.exports.getChannelFromVersion = (version) => {
  const splitVersion = version.split('-');
  return splitVersion[1] || null;
};
