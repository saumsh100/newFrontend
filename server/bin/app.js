
require('../config/initializeCodeTranspiler');

const globals = require('../config/globals');
const applyWebpack = require('../config/webpack/applyWebpack');
const handleErrors = require('../middleware/handleErrors');
const chalk = require('chalk');
const app = require('../config/express');

// require('../config/kue');

// Uses the NODE_ENV to determine logging full stack or not
app.set('showStackError', true);

// Load in webpack configurations
applyWebpack(app);


// Set the Handlebars templating engine
app.set('views', `${globals.root}/views`);
app.engine('hbs', require('../config/handlebars').engine);
app.set('view engine', 'hbs');

// Extra logging for communication with server
app.use((req, res, next) => {
  console.log(chalk.blue(req.method, req.originalUrl));
  next();
});


// Bind the routes

app.use(require('../routes'));

// Catch errors, log and respond to client
app.use(handleErrors);

// assume 404 since no middleware responded
/*app.use((req, res) => {
  res.status(404).render('404', {
    url: req.originalUrl,
    error: 'Not found',
  });
});*/

module.exports = app;
