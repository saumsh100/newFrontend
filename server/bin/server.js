
const http = require('http');
const globals = require('../config/globals');
const app = require('./app');
const Availability = require('../models/Availability');
const Patient = require('../models/Patient');

const server = http.createServer(app);

const io = require('socket.io')(server);

// TODO: implement Thinky!!!
io.on('connection', (socket) => {
  console.log('a new socket connected');
  socket.on('fetchAvailabilities', () => {
    Availability.run()
      .then((availabilities) => {
        socket.emit('receiveAvailabilities', availabilities);
      });
  });
  
  socket.on('fetchPatients', () => {
    Patient.run()
      .then((patients) => {
        socket.emit('receivePatients', patients);
      });
  });
  
  socket.on('addAvailability', (data) => {
    new Availability(data).save().then((availability) => {
      io.sockets.emit('availabilityAdded', availability);
    });
  });
  
  socket.on('removeAvailability', (data) => {
    Availability.get(data.id).then((availability) => {
      availability.delete()
        .then((result) => {
          io.sockets.emit('availabilityRemoved', result);
        });
    });
  });
  
  socket.on('syncClientTrigger', (data) => {
    console.log('sync client triggers!', 'data:', data);
  });
});

server.listen(globals.port, () => {
  console.log(`CareCru HTTP Server is running on port ${globals.port}`);
});
