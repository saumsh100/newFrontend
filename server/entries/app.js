
import '../../client/apps/default';
import '../../client/apps/Dashboard';
