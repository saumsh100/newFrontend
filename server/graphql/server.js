
import { Router } from 'express';
import graphQLHTTP from 'express-graphql';
import { sequelizeAuthMiddleware } from '../middleware/auth';
import schema from './data/schema';

const GraphQLRouter = Router();

const PORT = 4000;

const GraphQLServer = graphQLHTTP({
  schema,
  graphiql: true,
  pretty: true,
  subscriptionsEndpoint: `ws://localhost:${PORT}/subscriptions`,
});

// Reusing auth middleware as the server is exposed the same way an API would.
GraphQLRouter.all('*', sequelizeAuthMiddleware, GraphQLServer);

export default GraphQLRouter;

