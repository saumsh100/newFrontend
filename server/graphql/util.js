
import { getOffsetWithDefault, offsetToCursor } from 'graphql-relay';

/**
 * A simple function that accepts an array and connection arguments, and returns
 * a connection object for use in GraphQL
 *
 * This is a fork of connectionFromArraySlice
 * from https://github.com/graphql/graphql-relay-js/blob/master/src/connection/arrayconnection.js#L65
 *
 * Diferente form the original, is this case we build the cursor and pageInfo based on args
 * whitout slicing the array.
 * That way we can pass the already sliced or filtered array to the function.
 *
 * For now, this only works with forward pagination.
 */
// eslint-disable-next-line import/prefer-default-export
export function connectionFromArrayWithoutSlice(array, args, meta) {
  const { after, first, last } = args;
  const { arrayLength } = meta;

  if (typeof first === 'number') {
    if (first < 0) {
      throw new Error('Argument "first" must be a non-negative integer');
    }
  }
  if (typeof last === 'number') {
    if (last < 0) {
      throw new Error('Argument "last" must be a non-negative integer');
    }
  }

  const afterOffset = getOffsetWithDefault(after, 0);

  const startOffset = Math.max(afterOffset, -1) + 1;
  const endOffset = startOffset + first;

  const edges = array.map((value, index) => ({
    cursor: offsetToCursor(startOffset + index),
    node: value,
  }));

  const firstEdge = edges[0];
  const lastEdge = edges[edges.length - 1];
  const lowerBound = after ? afterOffset + 1 : 0;
  const upperBound = arrayLength;

  return {
    edges,
    pageInfo: {
      startCursor: firstEdge ? firstEdge.cursor : null,
      endCursor: lastEdge ? lastEdge.cursor : null,
      hasPreviousPage: typeof last === 'number' ? startOffset > lowerBound : false,
      hasNextPage: typeof first === 'number' ? endOffset < upperBound : false,
    },
  };
}
