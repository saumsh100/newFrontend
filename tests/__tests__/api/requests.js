
import request from 'supertest';
import app from '../../../server/bin/app';
import generateToken from '../../util/generateToken';
import { Request } from '../../../server/models';
import wipeModel, { wipeAllModels } from '../../util/wipeModel';
import { accountId, seedTestUsers } from '../../util/seedTestUsers';
import { serviceId, seedTestService } from '../../util/seedTestServices';
import { patientUserId, seedTestPatients } from '../../util/seedTestPatients';
import { appointmentId, seedTestAppointments } from '../../util/seedTestAppointments';
import { omitPropertiesFromBody } from '../../util/selectors';

const requestId = '272d86fc-f743-4cd6-b0c8-7906959bcc9f';
const requestSeed = {
  id: requestId,
  startDate: '2017-07-19T00:16:30.932Z',
  endDate: '2017-07-19T00:17:30.932Z',
  patientUserId,
  accountId,
  serviceId,
  createdAt: '2017-07-19T00:14:30.932Z',
};

async function seedTestRequest() {
  await seedTestAppointments();
  await seedTestService();
  await seedTestPatients();
  await wipeModel(Request);
  Request.save(requestSeed);
}

describe('/api/requests', () => {
  // Seed with some standard user data
  let token = null;
  beforeAll(async () => {
    await seedTestUsers();
    token = await generateToken({ username: 'manager@test.com', password: '!@CityOfBudaTest#$' });
  });

  afterAll(async () => {
    await wipeAllModels();
  });

  describe('GET /', () => {
    beforeEach(async () => {
      await seedTestRequest();
    });

    test('get all requests', () => {
      return request(app)
        .get('/api/requests')
        .set('Authorization', `Bearer ${token}`)
        .send({
          accountId,
        })
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('get all requests with patientUsers and services', () => {
      return request(app)
        .get('/api/requests?join=patientUser,service')
        .set('Authorization', `Bearer ${token}`)
        .send({
          accountId,
        })
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body, ['password']);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('POST /', () => {
    beforeEach(async () => {
      await wipeModel(Request);
    });

    test('create request', () => {
      return request(app)
        .post('/api/requests')
        .set('Authorization', `Bearer ${token}`)
        .send(requestSeed)
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('PUT /', () => {
    beforeEach(async () => {
      await seedTestRequest();
    });

    test('/:requestId - update request', () => {
      return request(app)
        .put(`/api/requests/${requestId}`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          endDate: '2017-07-19T00:18:30.932Z',
        })
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    test('/:requestId - reject request', () => {
      return request(app)
        .put(`/api/requests/${requestId}/reject`)
        .set('Authorization', `Bearer ${token}`)
        .send({
          accountId,
        })
        .expect(201)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });

    // TODO: Ask someone if this response is supposed to be an empty object
    test('/:requestId/confirm/:appointmentId - send confirmed request email', () => {
      return request(app)
        .put(`/api/requests/${requestId}/confirm/${appointmentId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(200)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });

  describe('DELETE /', () => {
    beforeEach(async () => {
      await seedTestRequest();
    });

    test('/:requestId - delete a request', () => {
      return request(app)
        .delete(`/api/requests/${requestId}`)
        .set('Authorization', `Bearer ${token}`)
        .expect(204)
        .then(({ body }) => {
          body = omitPropertiesFromBody(body);
          expect(body).toMatchSnapshot();
        });
    });
  });
});

