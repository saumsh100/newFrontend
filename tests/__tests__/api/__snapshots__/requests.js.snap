// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/requests DELETE / /:requestId - delete a request 1`] = `
Object {
  "entities": Object {},
}
`;

exports[`/api/requests GET / get all requests 1`] = `
Object {
  "entities": Object {
    "requests": Object {
      "272d86fc-f743-4cd6-b0c8-7906959bcc9f": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "appointmentId": null,
        "chairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "endDate": "2017-07-19T00:17:30.932Z",
        "id": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
        "insuranceCarrier": "test",
        "insuranceGroupId": null,
        "insuranceMemberId": "53245",
        "isCancelled": false,
        "isConfirmed": false,
        "isDeleted": false,
        "isSyncedWithPms": false,
        "note": "a standard note",
        "patientId": null,
        "patientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "pmsId": null,
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "requestingPatientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "sentRecallId": null,
        "serviceId": "c5beec65-73a0-4b58-ba48-65986931d054",
        "startDate": "2017-07-19T00:16:30.932Z",
        "suggestedChairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "suggestedPractitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
      },
    },
  },
  "result": Array [
    "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
  ],
}
`;

exports[`/api/requests GET / get all requests with patientUsers and services 1`] = `
Object {
  "entities": Object {
    "patientUsers": Object {
      "b8665c97-1e98-41ac-bf13-090b742de400": Object {
        "birthDate": "1996-07-19T00:14:30.932Z",
        "email": "testpatientuser@test.com",
        "firstName": "Jack",
        "gender": null,
        "id": "b8665c97-1e98-41ac-bf13-090b742de400",
        "insuranceCarrier": null,
        "insuranceGroupId": null,
        "insuranceMemberId": null,
        "isEmailConfirmed": false,
        "isPhoneNumberConfirmed": false,
        "lastName": "Daniels",
        "patientUserFamilyId": null,
        "phoneNumber": "+16049999999",
      },
    },
    "requests": Object {
      "272d86fc-f743-4cd6-b0c8-7906959bcc9f": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "appointmentId": null,
        "chairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "endDate": "2017-07-19T00:17:30.932Z",
        "id": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
        "insuranceCarrier": "test",
        "insuranceGroupId": null,
        "insuranceMemberId": "53245",
        "isCancelled": false,
        "isConfirmed": false,
        "isDeleted": false,
        "isSyncedWithPms": false,
        "note": "a standard note",
        "patientId": null,
        "patientUser": "b8665c97-1e98-41ac-bf13-090b742de400",
        "pmsId": null,
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "requestingPatientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "sentRecallId": null,
        "service": "c5beec65-73a0-4b58-ba48-65986931d054",
        "serviceId": "c5beec65-73a0-4b58-ba48-65986931d054",
        "startDate": "2017-07-19T00:16:30.932Z",
        "suggestedChairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "suggestedPractitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
      },
    },
    "services": Object {
      "c5beec65-73a0-4b58-ba48-65986931d054": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "bufferTime": null,
        "duration": 4,
        "id": "c5beec65-73a0-4b58-ba48-65986931d054",
        "isDefault": false,
        "isHidden": false,
        "name": "Test Service",
        "pmsId": null,
        "unitCost": null,
      },
    },
  },
  "result": Array [
    "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
  ],
}
`;

exports[`/api/requests GET / get all unSynced requests with patientUsers 1`] = `
Object {
  "data": Array [
    Object {
      "attributes": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "endDate": "2017-07-19T00:17:30.932Z",
        "formattedNote": "Date Requested: 5:14PM, 18/07/17
Reason for Booking: Test Service
Preferred Practitioner: Colonel Sanders
Note: a standard note

Patient Name: Jack Daniels
Phone Number: (604) 999-9999
Email: testpatientuser@test.com
Birth Date: 18/07/1996

Requested By: Self

Insurance Plan: test
Insurer Name: null
Member Id: 53245
Group Id: none",
        "insuranceCarrier": "test",
        "insuranceMemberId": "53245",
        "isDeleted": false,
        "note": "a standard note",
        "patientId": null,
        "pmsId": null,
        "startDate": "2017-07-19T00:16:30.932Z",
        "suggestedChairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "suggestedPractitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
      },
      "id": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
      "type": "appointmentRequest",
    },
  ],
}
`;

exports[`/api/requests POST / create request 1`] = `
Object {
  "entities": Object {
    "requests": Object {
      "272d86fc-f743-4cd6-b0c8-7906959bcc9f": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "appointmentId": null,
        "chairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "endDate": "2017-07-19T00:17:30.932Z",
        "id": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
        "insuranceCarrier": "test",
        "insuranceGroupId": null,
        "insuranceMemberId": "53245",
        "isCancelled": false,
        "isConfirmed": false,
        "isSyncedWithPms": false,
        "note": "a standard note",
        "patientId": null,
        "patientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "pmsId": null,
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "requestingPatientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "sentRecallId": null,
        "serviceId": "c5beec65-73a0-4b58-ba48-65986931d054",
        "startDate": "2017-07-19T00:16:30.932Z",
        "suggestedChairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "suggestedPractitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
      },
    },
  },
  "result": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
}
`;

exports[`/api/requests PUT / /:requestId - reject request 1`] = `
Object {
  "entities": Object {
    "requests": Object {
      "272d86fc-f743-4cd6-b0c8-7906959bcc9f": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "appointmentId": null,
        "chairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "endDate": "2017-07-19T00:17:30.932Z",
        "id": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
        "insuranceCarrier": "test",
        "insuranceGroupId": null,
        "insuranceMemberId": "53245",
        "isCancelled": true,
        "isConfirmed": false,
        "isSyncedWithPms": false,
        "note": "a standard note",
        "patientId": null,
        "patientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "pmsId": null,
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "requestingPatientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "sentRecallId": null,
        "serviceId": "c5beec65-73a0-4b58-ba48-65986931d054",
        "startDate": "2017-07-19T00:16:30.932Z",
        "suggestedChairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "suggestedPractitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
      },
    },
  },
  "result": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
}
`;

exports[`/api/requests PUT / /:requestId - update request 1`] = `
Object {
  "entities": Object {
    "requests": Object {
      "272d86fc-f743-4cd6-b0c8-7906959bcc9f": Object {
        "accountId": "62954241-3652-4792-bae5-5bfed53d37b7",
        "appointmentId": null,
        "chairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "endDate": "2017-07-19T00:18:30.932Z",
        "id": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
        "insuranceCarrier": "test",
        "insuranceGroupId": null,
        "insuranceMemberId": "53245",
        "isCancelled": false,
        "isConfirmed": false,
        "isSyncedWithPms": false,
        "note": "a standard note",
        "patientId": null,
        "patientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "pmsId": null,
        "practitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
        "requestingPatientUserId": "b8665c97-1e98-41ac-bf13-090b742de400",
        "sentRecallId": null,
        "serviceId": "c5beec65-73a0-4b58-ba48-65986931d054",
        "startDate": "2017-07-19T00:16:30.932Z",
        "suggestedChairId": "24d4e661-1455-4494-8fdb-c4ec0ddf804d",
        "suggestedPractitionerId": "497ff59a-4bae-4013-bdce-b6b5be91a1f5",
      },
    },
  },
  "result": "272d86fc-f743-4cd6-b0c8-7906959bcc9f",
}
`;

exports[`/api/requests PUT / /:requestId/confirm/:appointmentId - send confirmed request email 1`] = `
Object {
  "entities": Object {},
}
`;
